tests/integration/conftest.py

```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env python3
+# Copyright 2021 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+from pathlib import Path
+
+import pytest
+from pytest_operator.plugin import OpsTest
+
+_path_to_built_charm = None
+
+
+@pytest.fixture(scope="module")
+async def charm_under_test(ops_test: OpsTest) -> Path:
+    """Charm used for integration testing."""
+    global _path_to_built_charm
```

Comment:
@rbarry82 without this, the charm is being built again and again.
cf. [grafana](https://github.com/rbarry82/grafana-operator/blob/e9f169517988060d110e7fba87ac01b948f5c969/tests/integration/conftest.py#L25)

Reply:
https://github.com/charmed-kubernetes/pytest-operator/issues/40

Reply:
The way I'm "cheating" it to build globals (without ugly `globals`) keywords in the integration test harness is actually with `pytest_configure` to find the "best" controller based on the projected workload (bundle mixed machine/sidecar, whatever).

`pytest_sessionstart` looks like it will do what you need, though.

There isn't an existing global event loop in this hook, but you can literally just:

```
def pytest_sessionstart():
    pytest.path_to_built_charm = asyncio.run(...)
```